proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,434 @@
+changes within [TBD] business days. If a pull request becomes stale because the
+contributor is no longer actively engaged or working on it for more than [TBD]
+days, it should be reviewed to see if it is suitable for the maintainers to
+finish the work or marked as stale and documented for another user to continue
+working on it.
+
+Once a pull request is complete with to the satisfaction of the maintainers, it
+is approved and can be merged and acknowledged as contribution from the
+community.
+
+The maintainers will prioritize completing PRs that have had a review over new
+review requests. If a pull request becomes stale but there is sufficient work to
+complete it, the maintainers will complete the pull request. If after review and
+the maintainers agree that the implementation is not as discussed or there is
+not significant demand or bandwidth to complete it, the issue may go back to the
+intake phase and be available for another user to contribute a change.
+
+If an issue has not been explicitly marked as being available for contribution,
+it should not be expected that a pull request will get a review.
+
+## Measuring success and making adjustments
+
+We acknowledge that the proposed timelines and queue sizes may need to be
+adjusted. We will quantify the process with metrics to determine if it is
+sustainable or if it can be improved. Specifically, we should measure:
+
+1. The number of issues in each state of the contribution process: available for
+   contribution, actively being contributed, ready for review.
+1. The number of popular and highly requested features.
+1. The time elapsed between each state of the contribution process: the start of
+   a contribution and when it is marked as ready for review, the time spent
+   waiting for review, and the time to accept or reject a contribution.
+1. The rate of acceptance and attrition for contributions.
+
+In addition, we will monitor qualitative feedback through comments left on
+issues and pull requests. We will review the feedback for common patterns to
+determine if an improvement to the documentation will alleviate contributor
+frustration. It may also indicate that we need to make changes to the process as
+well.
+
+## Managing the existing backlog
+
+As of 2021-11-02, there are 189 pull requests and 465 issues in the AWS CLI
+GitHub repository. 52 pull requests and 81 issues have not been updated in more
+than a year. 151 issues have upvotes or reactions from five or more users. [TBD]
+of the open pull requests are extremely outdated and would require rebasing,
+which complicates the process of accepting them. [TBD] of the open pull requests
+are directly related to an existing feature request.
+
+Based on these numbers, we propose to close all issues that:
+
+1. Do not have at least 5 upvotes,
+1. Need a rebase that would require significant developer effort, and
+1. Do not have an open issue discussing the proposed change.
```

Comment:
I think we should add the criteria that maintainers have signed off on the approach too.

Reply:
agree, also close things that aren't passing the test suite.

Reply:
actually, I think stating that they need to meet the bar for 'ready for contrib' or 'ready for review' is sufficent and should not be reproduced here.